scripts/end2end_test.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+
+set -x # debugging
+
+DOCKER_COMPOSE_VERSION="1.8.1"
+DOCKER_COMPOSE_EXEC="/tmp/docker-compose"
+DOCKER_COMPOSE_FILE="docker/docker-compose.yml"
+
+curl -L "https://github.com/docker/compose/releases/download/${DOCKER_COMPOSE_VERSION}/docker-compose-Linux-x86_64" > ${DOCKER_COMPOSE_EXEC}
+chmod +x ${DOCKER_COMPOSE_EXEC}
+
+# start all docker containers
+${DOCKER_COMPOSE_EXEC} -f ${DOCKER_COMPOSE_FILE} up -d
+
+# wait for carbon input before sending data
+export WAIT_HOSTS="127.0.0.1:2003"
+export WAIT_TIMEOUT=120
+export METRICS_PER_SECOND=1000
+scripts/wait_for_endpoint.sh scripts/generate_test_data.sh start
+
+# give fakemetrics some warmup time
+sleep 30
+
+# verify the metrics have arrived in graphite and keep exit status
+scripts/verify_metrics_received.py 127.0.0.1 8080 10 ${METRICS_PER_SECOND}
+RESULT=${?}
```

Comment:
can just use `RESULT=$?` here. in fact, same comment about `${}` in lots of places. `${name}` is only needed when you have something like `${name}otherstr or `${name}-other-string` where `$name-other-str would` be an incorrect variable name, but in most cases (eg. variable followed by whitespace) this is not needed